src/main/java/com/eternalcode/parcellockers/parcel/ParcelRepositoryJdbcImpl.java

```diff
@@ -0,0 +1,119 @@
+            }
+            return Optional.empty();
+        });
+    }
+
+    @Override
+    public CompletableFuture<Set<Parcel>> findAll() {
+        return CompletableFuture.supplyAsync(() -> {
+            ParcelLockerRepositoryJdbcImpl parcelLockerRepository = ParcelLockerRepositoryJdbcImpl.create(this.jdbcConnectionProvider);
+            Set<Parcel> parcels = new HashSet<>();
+
+            try (ResultSet resultSet = this.jdbcConnectionProvider.executeQuery("SELECT * FROM `parcels`")) {
+                while (resultSet.next()) {
+                    Parcel parcel = this.extractParcel(parcelLockerRepository, resultSet);
+                    parcels.add(parcel);
+                }
+            }
+            catch (SQLException exception) {
+                throw new RuntimeException(exception);
+            }
+            return parcels;
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public CompletableFuture<Void> remove(Parcel parcel) {
+        return CompletableFuture.runAsync(() -> {
+            this.jdbcConnectionProvider.executeUpdate("DELETE FROM `parcels` WHERE `uuid` = " + parcel.getUuid().toString());
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public CompletableFuture<Void> remove(UUID uuid) {
+        return CompletableFuture.runAsync(() -> {
+            this.jdbcConnectionProvider.executeUpdate("DELETE FROM `parcels` WHERE `uuid` = " + uuid.toString());
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    public static ParcelRepositoryJdbcImpl create(JdbcConnectionProvider jdbcConnectionProvider) {
+        jdbcConnectionProvider.executeUpdate("CREATE TABLE IF NOT EXISTS `parcels` (`uuid` VARCHAR(36) NOT NULL," +
+                " `name` VARCHAR(24) NOT NULL," +
+                " `description` VARCHAR(64)," +
+                " `priority` VARCHAR(5) NOT NULL," +
+                " `receiver` VARCHAR(36) NOT NULL," +
+                " `size` VARCHAR(10) NOT NULL," +
+                " `entryLocker` VARCHAR(36) NOT NULL," +
+                " `destinationLocker` VARCHAR(36) NOT NULL," +
+                " `sender` VARCHAR(36) NOT NULL," +
+                "  PRIMARY KEY (`uuid`))");
+        return new ParcelRepositoryJdbcImpl(jdbcConnectionProvider);
+    }
+
+    private Parcel extractParcel(ParcelLockerRepositoryJdbcImpl parcelLockerRepository, ResultSet resultSet) throws SQLException {
+        ParcelMeta meta = new ParcelMeta(
+                resultSet.getString("name"),
+                resultSet.getString("description"),
+                resultSet.getBoolean("priority"),
+                UUID.fromString(resultSet.getString("receiver")),
+                ParcelSize.valueOf(resultSet.getString("size")),
+                parcelLockerRepository.findByUuid(UUID.fromString(resultSet.getString("entryLocker"))).join().get(),
+                parcelLockerRepository.findByUuid(UUID.fromString(resultSet.getString("destinationLocker"))).join().get());
```

Comment:
nie powinieneś w taki sposób użyć chyba completabla, a jeśli dajesz już `join` to daj `timeout`

Reply:
Najlepiej było by zastosować to w metodzie `findByUuid`, ale przy użyciu `Optional<Parcel/ParcelLocker>`, orTimeout zwraca `Optional<?>`, co wywala błąd.